### ISSUE TITLE
Real-valued (Medical) Time Series Generation with Recurrent Conditional GANs

### ISSUE THREAD

**Issue by agitter** [greenelab/deep-review#539]
https://arxiv.org/abs/1706.02633

> Generative Adversarial Networks (GANs) have shown remarkable success as a framework for training models to produce realistic-looking data. In this work, we propose a Recurrent GAN (RGAN) and Recurrent Conditional GAN (RCGAN) to produce realistic real-valued multi-dimensional time series, with an emphasis on their application to medical data. RGANs make use of recurrent neural networks in the generator and the discriminator. In the case of RCGANs, both of these RNNs are conditioned on auxiliary information. We demonstrate our models in a set of toy datasets, where we show visually and quantitatively (using sample likelihood and maximum mean discrepancy) that they can successfully generate realistic time-series. We also describe novel evaluation methods for GANs, where we generate a synthetic labelled training dataset, and evaluate on a real test set the performance of a model trained on the synthetic data, and vice-versa. We illustrate with these metrics that RCGANs can generate time-series data useful for supervised training, with only minor degradation in performance on real test data. This is demonstrated on digit classification from 'serialised' MNIST and by training an early warning system on a medical dataset of 17,000 patients from an intensive care unit. We further discuss and analyse the privacy concerns that may arise when using RCGANs to generate realistic synthetic medical time series data.

Relevant to the review for the medical time series data.  From the paper:
> ...we based our work on the recently-released Philips eICU database. This dataset was collected by the critical care telehealth program provided by Philips. It contains around 200,000 patients from 208 care units across the US... we focus on generating the four most frequently recorded, regularly-sampled variables measured by bedside monitors: oxygen saturation measured by pulse oximeter (SpO2), heart rate (HR), respiratory rate (RR) and mean arterial pressure (MAP)

Their privacy discussion centers on whether the generated data is too similar to the training data such that an adversary could recover the training data.  @cgreene I'm curious if you think this is a strong enough privacy test.

**Comment by cgreene** [comment]
We have done both of those evaluations (RF & NN) for the approach that @brettbj has been working on. From our experience, those are not directly aligned with good performance with privacy. I think they are helpful evaluations, but I'm not sure that they are sufficient. We debated putting out a paper on that last year and held off b/c we wanted to make it reproducible and worried that putting out the models might leak data. I guess I would say - I can't conclusively show that the approach leaks training data, but since we couldn't say anything conclusive about the type of privacy that it preserves we elected not to risk it.

**Comment by agitter** [comment]
Interesting.  So is this worth adding during revisions?

**Comment by cgreene** [comment]
Yea. I am hoping that we can get @brettbj's preprint up too, and then we can deal with all of them together. That section will be much improved though. I think this paper is a more interesting contribution than the one we're already talking about.
